lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -1731,9 +1752,21 @@ optimizeBridgedSwiftToObjCCast(SILInstruction *Inst,
       }
       break;
     case ParameterConvention::Direct_Unowned:
+      assert(!AddressOnlyType &&
+             "AddressOnlyType with Direct_Unowned is not supported");
       break;
-    case ParameterConvention::Indirect_In:
     case ParameterConvention::Indirect_In_Guaranteed:
+      // Source as-is, we don't need to copy it due to guarantee
+      break;
+    case ParameterConvention::Indirect_In: {
+      assert(AddressOnlyType &&
+             "Expected an AddressOnlyType with Indirect_In Convention");
```

Comment:
The parameter may still be indirect even if the incoming type isn't address only. This happens with generic types like `Optional`, for instance, since `Optional` may be address-only in the generic case, but is loadable when instantiated on a loadable type. It may be better to check the parameter convention of the function instead of the loadability of the type when deciding whether to load it.